.github/workflows/release.yml

```diff
@@ -1,63 +1,112 @@
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install cibuildwheel
+        run: pipx install cibuildwheel==2.13.1
+      - id: set-matrix
+        run: |
+          MATRIX=$(
+            {
+              cibuildwheel --print-build-identifiers --platform linux \
+              | jq -nRc '{"only": inputs, "os": "ubuntu-latest"}' \
+              && cibuildwheel --print-build-identifiers --platform macos \
+              | jq -nRc '{"only": inputs, "os": "macos-latest"}' \
+              && cibuildwheel --print-build-identifiers --platform windows \
+              | jq -nRc '{"only": inputs, "os": "windows-2019"}'
+            } | jq -sc
+          )
+          echo "include=$MATRIX" >> $GITHUB_OUTPUT
+        env:
+          CIBW_BUILD: "cp39-* cp310-* cp311-*"
+          # Skip 32 bit windows builds and musllinux due to lack of numpy wheels
+          CIBW_SKIP: "*-win32 *-musllinux*"
+          CIBW_ARCHS_MACOS: x86_64 arm64
+  build_wheels:
+    name: Build ${{ matrix.os }} ${{ matrix.only }}
+    needs: generate-wheels-matrix
+    strategy:
+      matrix:
+        include: ${{ fromJson(needs.generate-wheels-matrix.outputs.include) }}
+    runs-on: ${{ matrix.os }}
     defaults:
       run:
-        shell: bash -l {0}
-
+        shell: bash
     steps:
       - uses: actions/checkout@v3
-      - uses: conda-incubator/setup-miniconda@v2
         with:
-          activate-environment: test-environment
-          python-version: ${{ matrix.python-version }}
-          channels: conda-forge/label/testing,conda-forge
+          # We need the full history to generate the proper version number
+          fetch-depth: 0
 
-      - name: Install dependencies
-        run: |
-          PACKAGES="cython fiona matplotlib-base numpy pyproj pykdtree scipy"
-          PACKAGES="$PACKAGES owslib pep8 pillow pyshp pytest"
-          PACKAGES="$PACKAGES pytest-xdist setuptools_scm shapely"
-          conda install $PACKAGES
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - uses: pypa/cibuildwheel@v2.13.1
+        with:
+          only: ${{ matrix.only }}
+
+      - uses: actions/upload-artifact@v3
+        with:
+          path: ./wheelhouse/*.whl
+
+  build_sdist:
+    name: Build source distribution
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # We need the full history to generate the proper version number
+          fetch-depth: 0
 
-      - name: Create sdist
-        run: python setup.py build_ext sdist
+      - uses: actions/setup-python@v4
+        name: Install Python
+        with:
+          python-version: '3.11'
 
-      - name: Save built packages as artifact
-        uses: actions/upload-artifact@v3
+      - name: Build sdist
+        run: |
+          python -m pip install build
+          # install compiles the cython *.pyx to *.cpp for us
+          python -m pip install .
```

Comment:
This doesn't seem right, though I don't know how to effectively Google `build` things.

Reply:
I think you're right here. See my above reply. I think what this was doing for me was to compile the `.pyx` to `.cpp`, so we need another way of doing that then if we don't want to pip install here before the build. Any thoughts/ideas appreciated on this.